### ISSUE TITLE
Local charm icons do not load with GiJoe

### ISSUE THREAD

**Issue by hatched** [juju/juju-gui#1682]
I deployed the gui charm locally into a juju2 model. Then updated and loaded the gui using `juju gui`. The gui charm on the canvas had a broken icon with a 404:

`https://user-admin:0842649783624fb987cef15ba621c504@10.21.146.54:17070/juju-core/charms?url=local:trusty/juju-gui-0&file=icon.svg`

Using FireFox

**Comment by frankban** [comment]
Yes I see why it fails. The GUI assumes it is talking to the GUI server. That's a GUI server URL (see the "/juju-core/" fragment) used to redirect to the Juju HTTP API (same URL without the the "/juju-core" prefix). So basically https://user-admin:0842649783624fb987cef15ba621c504@10.21.146.54:17070/charms?url=local:trusty/juju-gui-0&file=icon.svg should get you the icon. One possible solution is to have an option for overriding the default HTTP API URL to be used in the charmed GUI scenario. But we really need to figure out an elegant solution here.

**Comment by wurde** [comment]
I think this is still an issue. I can't seem to get my local charms to render their icon.svg.

2.0-beta8-xenial-amd64

**Comment by wurde** [comment]
The offending code: [service.js#-1617-1629](https://github.com/juju/juju-gui/blob/bacd8fcea43dd55943731101e454147f1d95d3aa/jujugui/static/gui/src/app/views/topology/service.js#-1617-1629
)

```javascript
node.append('image')
  .classed('service-icon', true)
  .attr({
    'xlink:href': function(d) {
      return d.icon;
    },
    width: 96,
    height: 96,
    transform: function(d) {
      return (d.subordinate ? 'translate(17, 17)' : 'translate(47, 47)');
    },
    'clip-path': function(d) { return 'url(#clip-mask)'; }
  });
```

**Comment by kadams54** [comment]
This issue seems to be resolved with the latest GUI and juju 2.0-beta17-trusty-amd64. @hatched and I are unable to replicate at the moment.
